Add issue templates (#2086)
vaadin/web-components: .github/ISSUE_TEMPLATE/bug_report.md

---
name: "\U0001F41B Bug Report"
about: "Please report issues related to the TypeScript and HTML API of Vaadin components here."
title: ''
labels: 'i: needs triage'
assignees: ''

---

<!--
  Please fill out each section below, otherwise, your issue risks taking an infinite amount of time to triage. This info allows the maintainers team to diagnose and fix your issue as quickly as possible.

  Before opening a new issue, please search existing issues: https://github.com/vaadin/web-components/issues.
-->

### Description

Describe the issue that you're seeing.

<!-- Example: The `vaadin-grid` element scrolls when applying filter. -->

### Expected outcome

> What should happen?

Reviewer: ```suggestion
What did you expect to happen?
```

<!-- Example: No scrolling should happen. -->

### Actual outcome

> What happened.

Reviewer: ```suggestion
What happened instead.
```

<!-- Example: Grid content occasionally gets scrolled. -->

### Live Demo

> A screenshot or video with issue reproduced.

Reviewer: This could be marked as optional IMO. I'd rather that the reporter invests time into a reproduction.

### Minimal reproducible example

> **Please** provide a code snippet with issue reproduction, this makes your issue _much_ easier to diagnose.

Reviewer: ```suggestion
**Please** provide a demo project or code snippet that reproduces the issue, this makes your issue _much_ easier to diagnose.
```
I would also like to be more forceful here by stating that issues without reproduction are unlikely to be worked on. The current wording could easily lead to a mindset of "Not my problem if it takes you longer to diagnose", while the reality is that we often can not do anything without a reproduction. Could also have a quick check with Rolf for a reasonable wording of this.

<!--
```
<vaadin-grid</vaadin-grid>
```
```
const grid = document.querySelector('vaadin-grid');

> grid.items = items;

Reviewer: Can we improve this to add a filter here to make the example consistent with the remaining examples?

Author: Adding `vaadin-grid-filter` would probably explode this example with too many details. Here I am trying to show how users can provide a code snippet rather than provide an explicit one.

If you think a filter element is absolutely needed, I can add it.

Reviewer: It's OK, I wasn't sure how much code it would add.

```
-->

### Steps to reproduce

> Clear steps describing how to reproduce the issue.

Reviewer: I'm thinking that this text could also refer to the reproduction that _should_ be provided above. 

The example below would then just be:
```
3. Open the page in a web browser.
4. Start typing in the filter input
```

Author: I would like to preserve the details for the reproduction if possible (with initial steps, like adding `grid` and `grid-filter` to the page). So the user will provide as many details as possible when creating `Steps to reproduce`.

Reviewer: OK sure.

<!-- Example
1. Put a `vaadin-grid` element in the page.
2. Add `vaadin-grid-filter` element to the column header template.
3. Open the page in a web browser.
4. Start typing in the filter input.
-->

### Environment

> - Version of the webcomponent / platform.

Reviewer: This might not be clear enough. For the most part we want to support Flow / Fusion users, so they should only need to specify the Vaadin platform version. For standalone web-component usage we would indeed need the web component version, but that is also aligned with the platform since 20. So maybe just mention the platform version here?

Author: I don't have strong preferences on this one but would like to show that users can specify webcomponent versions for the components used in 14.x where the versions are not aligned.

Reviewer: OK, one small improvement I would suggest is to use the term `Vaadin platform`

Author: Sounds good! Updated.
